Allow custom IDs

Xstoudi opened the pull request:
Should resolve #25

stropitek commented:
There is an issue in the diff. All the lines of the files you modified have changed.

targos commented on src/Model/__tests__/Model.test.ts:
```diff
@@ -1,294 +1,307 @@
+    _id: 'test',
+    username: nextUsername(),
+    password: 'mypass',
+  } as any);
```
why do you need `any` here?

Xstoudi commented:
I fixed the type to allow you to pass _id without passing any, tell me if you consider it isn't the best way to do it.

For the line diff issue, I fixed it too.

targos commented:
For `_id`, we can just define it on the Model directly (public readonly _id) like in the public types: https://github.com/zakodium/adonis-mongodb/blob/210cacdd1f61d512bf58155578c7a21fa85d7f3c/src/types.ts#L149

Xstoudi commented:
It's already the case in the types.ts so it's only tests problem.
I added the field as any, as there is no type-concern in the Model.
